src/utils.ts

```diff
@@ -0,0 +1,63 @@
+// Import types and APIs from graph-ts
+import {
+  BigInt,
+  ByteArray,
+  EthereumEvent,
+} from '@graphprotocol/graph-ts'
+// Import entity types generated from the GraphQL schema
+import { Domain, Registration } from './types/schema'
+
+export function createEventID(event: EthereumEvent): string {
+  return event.block.number.toString().concat('-').concat(event.logIndex.toString())
+}
+
+export const ROOT_NODE = '0x0000000000000000000000000000000000000000000000000000000000000000'
+export const EMPTY_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+export function loadOrCreateDomain(node:string): Domain {
```

Comment:
I'm generally a fan of the deduplication, but these loadOrCreate methods are only used in one file each, so they should stay in those files.